documentation_contributions: show how to set up a venv and remove outdated notes

gotmax23 opened the pull request:
This includes some updates to the documentation_contributions doc that were split out from https://github.com/ansible/ansible-documentation/pull/707. See this PR's commit messages for more details.

gotmax23 committed:
documentation_contributions: show how to set up a venv

The current steps that advise installing docs dependencies in a user's
global environment are a very bad idea. They won't work on newer
versions of Debian that have adopted PEP 668 that disables `pip install
--user`. On systems where it does work, it's likely to pollute or
entirely break developers' global Python environment.

gotmax23 committed:
documentation_contributions: remove outdated notes

oraNod commented on docs/docsite/rst/community/documentation_contributions.rst:
```diff
@@ -91,13 +91,13 @@ To build documentation locally, ensure you have a working :ref:`development envi
 
-      pip install --user --upgrade pip
+      python3 -m venv ./venv
+      . ./venv/bin/activate
```
```suggestion
      source ./venv/bin/activate
```

oraNod replied:
Using `source ./venv` might not be as efficient but it's probably more obvious. I'd be concerned that some users might not really spot the first dot character in `. ./venv`. Just seems like it's a bit more likely to go wrong somehow as dot is a bit less readable.

samccann replied:
fwiw  I had to read dons comment twice and the pr change to see what he was talking about.. which to me means yeah, I didn't notice the first dot character :-)

oraNod commented on docs/docsite/rst/community/documentation_contributions.rst:
```diff
@@ -91,13 +91,13 @@ To build documentation locally, ensure you have a working :ref:`development envi
-Drop the ``--user`` option in the following commands if you use a virtual environment (venv/virtenv).
 
-#. Upgrade pip before installing dependencies (recommended).
+#. Set up a virtual environment in which to install the dependencies
```
```suggestion
#. Set up a virtual environment in which to install dependencies.
```

gotmax23 committed:
documentation_contributions: improve sentence structure

Co-authored-by: Don Naro <dnaro@redhat.com>

gotmax23 committed:
documentation_contributions: use `source` instead of `.`

`source` is not part of POSIX sh, but I most people are using bash or zsh or whatever which does have it.

Co-authored-by: Don Naro <dnaro@redhat.com>

oraNod commented on docs/docsite/rst/community/documentation_contributions.rst:
```diff
@@ -109,24 +109,9 @@ Drop the ``--user`` option in the following commands if you use a virtual enviro
-
-.. note::
-
-  	After checking out ``ansible/ansible-documentation``, make sure the ``docs/docsite/rst`` directory has strict enough permissions. It should only be writable by the owner's account. If your default ``umask`` is not 022, you can use ``chmod go-w docs/docsite/rst`` to set the permissions correctly in your new branch.  Optionally, you can set your ``umask`` to 022 to make all newly created files o ...
```
Hmm... I'm all for trimming out content that is no longer needed but I wonder if this is somehow losing a bit of history and worry it might confuse some users. What if we rephrased this to make it stand out as deprecated? That way we could perhaps give it a release cycle or so before completely dropping it.

gotmax23 replied:
Are you referring to the part about manually installing Python packages (we use venv+pip for that now) and manually installing devel tools (everything we need should have pre-built wheels on PyPI) or the part about setting permissions?

oraNod replied:
Both. Maybe users don't want to use a venv - or weren't using a venv previously - and that list of packages might still be relevant.

I guess with the move from `ansible/ansible` the part about permissions is no longer relevant. Maybe it's likely no users will ever hit this again and I'm being overly cautious about dropping it. And it might not make that much sense to point out something you used to need to do but now don't.

gotmax23 replied:
Even for developers not using venvs[^1], the information about prerequisites is entirely out of date. `gcc`, `libyaml`, and `make` shouldn't be necessary due to the aforementioned pre-compiled wheels. Newer versions of pip build wheels in an isolated environment and don't need `wheel` installed manually. `six` and `pyparsing` shouldn't need to be installed manually, and the latter is no longer part of our dependency tree.

[^1]: which is a bad idea. Installing a lockfile into a global environment is quite likely to break things

oraNod replied:
yeah I take the point about not using venvs and am not trying to argue too vehemently about retaining this detail. if you feel removing it is the right approach, I have more than enough faith. my only concern is that we could have users who could still find these details useful.

oraNod reviewed (approved):
Thanks @gotmax23

gotmax23 commented:
Thanks everyone!
